diff --git a/src/java.base/share/classes/sun/security/ssl/TrustStoreManager.java b/src/java.base/share/classes/sun/security/ssl/TrustStoreManager.java
index 8f464e058e2..795a14bd4f4 100644
--- a/src/java.base/share/classes/sun/security/ssl/TrustStoreManager.java
+++ b/src/java.base/share/classes/sun/security/ssl/TrustStoreManager.java
@@ -68,6 +68,7 @@ public static KeyStore getTrustedKeyStore() throws Exception {
      *
      * The preference of the default trusted KeyStore is:
      *    javax.net.ssl.trustStore
+     *    system environment variable JAVAX_NET_SSL_TRUSTSTORE
      *    jssecacerts
      *    cacerts
      */
@@ -125,7 +126,7 @@ private TrustStoreDescriptor(String storeName, String storeType,
         static TrustStoreDescriptor createInstance() {
             // Get the system properties for trust store.
             String storePropName = System.getProperty(
-                    "javax.net.ssl.trustStore", jsseDefaultStore);
+                    "javax.net.ssl.trustStore", System.getenv("JAVAX_NET_SSL_TRUSTSTORE"));
             String storePropType = System.getProperty(
                     "javax.net.ssl.trustStoreType",
                     KeyStore.getDefaultType());
@@ -137,6 +138,9 @@ static TrustStoreDescriptor createInstance() {
             String temporaryName = "";
             File temporaryFile = null;
             long temporaryTime = 0L;
+            if (storePropName == null) {
+                storePropName = jsseDefaultStore;
+            }
             if (!"NONE".equals(storePropName)) {
                 String[] fileNames =
                         new String[] {storePropName, defaultStore};
